crates/egui/src/menu.rs

```diff
@@ -288,7 +288,23 @@ impl MenuRoot {
         {
             // menu not open and button clicked
             // or button hovered while other menu is open
-            let pos = response.rect.left_bottom();
+            drop(input);
+
+            let mut pos = response.rect.left_bottom();
+            if let Some(root) = root.inner.as_mut() {
+                let menu_rect = root.menu_state.read().rect;
+                let screen_rect = response.ctx.input().screen_rect;
+
+                if pos.y + menu_rect.height() > screen_rect.max.y {
+                    pos.y = screen_rect.max.y - menu_rect.height();
+                    pos.x += response.rect.width();
```

Comment:
I guess this is so that the menu doesn't cover the button, but wouldn't it make more sense to move the menu up by another `response.rect.height()`, so the menu is now above the button? That way a menu-button at the bottom of the screen will have its menu open above it, which makes sense to me.

Reply:
It does make sense, I fixed it.